src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,616 @@
+                {
+                case EVENT_GHOST_VISUAL:
+                    DoCast(me, SPELL_GHOST_DISGUISE);
+                    _events.ScheduleEvent(EVENT_REMOVE_GHOST_VISUAL, 6000, 0, PHASE_TWO);
+                    break;
+                case EVENT_REMOVE_GHOST_VISUAL:
+                    me->RemoveAura(SPELL_GHOST_DISGUISE);
+                    _events.ScheduleEvent(EVENT_GHOST_VISUAL, 9000, 0, PHASE_TWO);
+                    break;
+                case EVENT_RESURFACE_SOON:
+                    Talk(EMOTE_EMERGE_SOON);
+                    _events.CancelEvent(EVENT_GHOST_VISUAL);
+                    _events.CancelEvent(EVENT_REMOVE_GHOST_VISUAL);
+                    break;
+                }
+            }
+        }
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_frozen_coreAI(creature);
+    }
+};
+
+class npc_ahune_bunny : public CreatureScript
+{
+public:
+    npc_ahune_bunny() : CreatureScript("npc_ahune_bunny") { }
+
+    struct npc_ahune_bunnyAI : public ScriptedAI
+    {
+        npc_ahune_bunnyAI(Creature* creature) : ScriptedAI(creature), _summons(me)
+        {
+            instance = me->GetInstanceScript();
+            count = 0;
+            say_timer = 3000;
+        }
+
+        InstanceScript* instance;
+        uint32 count;
+        uint64 say_timer;
+        SummonList _summons;
+
+        void JustSummoned(Creature* summon) override
+        {
+            _summons.Summon(summon);
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            _summons.DespawnAll();
+        }
+
+        void DoAction(int32 action)
+        {
+            if (action = ACTION_START_EVENT)
+            {
+                TempSummon* sliperryFloor = me->SummonCreature(NPC_SLIPPERY_FLOOR, SummonPositions[7], TEMPSUMMON_MANUAL_DESPAWN);
+                sliperryFloor->SetReactState(REACT_PASSIVE);
+                sliperryFloor->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_DISABLE_MOVE);
+                sliperryFloor->AddAura(SPELL_SLIPPERY_FLOOR_AMBIENT, sliperryFloor);
+                for (int counter = 1; counter < 7; counter++)
+                {
+                    if (counter < 4)
+                        me->SummonCreature(NPC_EARTHEN_RING_TOTEM, SummonPositions[counter], TEMPSUMMON_MANUAL_DESPAWN);
+                    else
+                        me->SummonCreature(NPC_EARTHEN_RING_FLAMECALLER, SummonPositions[counter], TEMPSUMMON_MANUAL_DESPAWN);
+                }
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (say_timer <= diff)
+            {
+                if (Player* player = SelectRandomPlayer(100.0f, false))
+                {
+                    if (count == 0)
+                        sCreatureTextMgr->SendChat(me, SAY_PLAYER_TEXT_1, NULL, CHAT_MSG_SAY, LANG_UNIVERSAL, TEXT_RANGE_NORMAL, 0, TEAM_OTHER, false, player);
```

Comment:
It's one of those cases where the text is said by the player, not creature. Remember when we looked over it the other day? Well, it's the same case here